File: php/testcode/benchmark_test_00731.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00731(BenchmarkRequest $req): BenchmarkResponse {
    $path = $req->param('path');
    if (str_contains($path, '://') || str_starts_with($path, '//')) {
        return BenchmarkResponse::badRequest('Absolute URLs not allowed');
    }
    header("Location: /" . ltrim($path, '/'));
    return BenchmarkResponse::ok('');
}

File: rust/testcode/benchmark_test_00731.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let user_dir = req.param("user_id");
    let filename = req.param("filename");
    let content = req.param("content");
    let path = format!("/uploads/{}/{}", user_dir, filename);
    let _ = std::fs::write(&path, content.as_bytes());
    super::shared::BenchmarkResponse::ok("Stored")
}
